reason-react-native/src/apis/Style.md

```diff
@@ -4,4 +4,1297 @@ title: Style
+
+```reason
+open ReactNative;
+
+let styles =
+  Style.(
+    StyleSheet.create({
+      "container":
+        viewStyle(
+          ~maxHeight=600.->pt,
+          ~width=800.->pct,
+          ~justifyContent=`flexStart,
+          ~alignItems=`center,
+          ~margin=auto,
+          (),
+        ),
+      "cornerThing":
+        viewStyle(
+          ~position=`absolute,
+          ~top=100.->pt,
+          ~right=(-20.)->pt,
+          ~transform=[|rotate(~rotate=4.->deg)|],
+          (),
+        ),
+      "text": textStyle(~textTransform=`uppercase, ()),
+    })
+  );
+
+[@react.component]
+let make = (~isSomething) => {
+  <View style=styles##container>
+    <View style=styles##cornerThing>
+      <Text
+        style=Style.(
+          arrayOption([|
+            Some(styles##text),
+            isSomething ? Some(style(~opacity=0.05, ())) : None,
+          |])
+        )
+      />
+    </View>
+  </View>;
+};
+```
+
+## Style Units
+
+### `size`
+
+`size` is used in various place to express logical pixels (`pt` function),
+percentage (`pct` function) & also `auto` (inline string, edge case for
+`margin`).
+
+As soon as `Style` module is open, you make `size` in various way as you can see
+in this random examples
+
+- `~height=10.5->pt` (= `~height=pt(10.5)`)
+- `~width=55.->pct` (= `~width=pct(55.)`)
+- `~margin=auto`
+
+### `offset`
+
+`offset` is used for `shadowOffset` & `textShadowOffset`.
+
+Eg: `~shadowOffset=offset(~height=2., ~width=4.)`
+
+### `angle`
+
+`angle` is used for transforms. It can be expressed in degrees (`deg` function)
+or in radians (`rad` function):
+
+- `20.->deg` (eg: `rotateX(~rotateX=20.->deg)`
+- `0.5.->rad` (eg: `rotateZ(~rotateZ=0.5->rad)`
+
+## Style Functions
+
+### `Style.style`
+
+For convenience, this function allows you to prepare a style object with all
+styles available in React Native. The nice side of this is that you can use this
+function & not think about what component is going to use it. On the other hand,
+this function can trigger React Native error screen (eg: if you pass `color` to
+a `View` component).
+
+`style` accepts all the Style Props:
+
+- [Layout styles](#layout-styles-props)
+- [Shadow styles](#shadow-styles-props)
+- [Transform styles](#transform-styles-props)
+- [View styles](#view-styles-props)
+- [Text styles](#text-styles-props)
+- [Image styles](#image-styles-props)
+
+⚠️ _Internally, React Native codebase types for this function are called
+`____DangerouslyImpreciseStyle_Internal`. That's why we created new functions:
+[`viewStyle`](#style-viewstyle), [`textStyle`](#style-textstyle) &
+[`imageStyle`](#style-imagestyle)_.
+
+### `Style.viewStyle`
+
+This function accept all React Native following styles
+
+- [Layout styles](#layout-styles-props)
+- [Shadow styles](#shadow-styles-props)
+- [Transform styles](#transform-styles-props)
+- [View styles](#view-styles-props)
+
+### `Style.textStyle`
+
+This function accept all React Native following styles
```

Comment:
subject-verb mismatch, and perhaps better to restate as (as above):
This function accepts all React Native styles below: